database/src/tables.rs

```diff
@@ -12,6 +13,184 @@ use crate::{
+    fn block_heights_mut(&mut self) -> &mut impl DatabaseRw<BlockHeights>;
+    fn key_images_mut(&mut self) -> &mut impl DatabaseRw<KeyImages>;
+    fn num_outputs_mut(&mut self) -> &mut impl DatabaseRw<NumOutputs>;
+    fn pruned_tx_blobs_mut(&mut self) -> &mut impl DatabaseRw<PrunedTxBlobs>;
+    fn prunable_hashes_mut(&mut self) -> &mut impl DatabaseRw<PrunableHashes>;
+    fn outputs_mut(&mut self) -> &mut impl DatabaseRw<Outputs>;
+    fn prunable_tx_blobs_mut(&mut self) -> &mut impl DatabaseRw<PrunableTxBlobs>;
+    fn rct_outputs_mut(&mut self) -> &mut impl DatabaseRw<RctOutputs>;
+    fn tx_ids_mut(&mut self) -> &mut impl DatabaseRw<TxIds>;
+    fn tx_heights_mut(&mut self) -> &mut impl DatabaseRw<TxHeights>;
+    fn tx_unlock_time_mut(&mut self) -> &mut impl DatabaseRw<TxUnlockTime>;
+}
+
+impl<A, B, C, D, E, F, G, H, I, J, K, L, M, N, O> TablesMut
+    for (A, B, C, D, E, F, G, H, I, J, K, L, M, N, O)
+where
+    A: DatabaseRw<BlockInfoV1s>,
+    B: DatabaseRw<BlockInfoV2s>,
+    C: DatabaseRw<BlockInfoV3s>,
+    D: DatabaseRw<BlockBlobs>,
+    E: DatabaseRw<BlockHeights>,
+    F: DatabaseRw<KeyImages>,
+    G: DatabaseRw<NumOutputs>,
+    H: DatabaseRw<PrunedTxBlobs>,
+    I: DatabaseRw<PrunableHashes>,
+    J: DatabaseRw<Outputs>,
+    K: DatabaseRw<PrunableTxBlobs>,
+    L: DatabaseRw<RctOutputs>,
+    M: DatabaseRw<TxIds>,
+    N: DatabaseRw<TxHeights>,
+    O: DatabaseRw<TxUnlockTime>,
+{
+    fn block_info_v1s_mut(&mut self) -> &mut impl DatabaseRw<BlockInfoV1s> {
+        &mut self.0
+    }
+    fn block_info_v2s_mut(&mut self) -> &mut impl DatabaseRw<BlockInfoV2s> {
+        &mut self.1
+    }
+    fn block_info_v3s_mut(&mut self) -> &mut impl DatabaseRw<BlockInfoV3s> {
+        &mut self.2
+    }
+    fn block_blobs_mut(&mut self) -> &mut impl DatabaseRw<BlockBlobs> {
+        &mut self.3
+    }
+    fn block_heights_mut(&mut self) -> &mut impl DatabaseRw<BlockHeights> {
+        &mut self.4
+    }
+    fn key_images_mut(&mut self) -> &mut impl DatabaseRw<KeyImages> {
+        &mut self.5
+    }
+    fn num_outputs_mut(&mut self) -> &mut impl DatabaseRw<NumOutputs> {
+        &mut self.6
+    }
+    fn pruned_tx_blobs_mut(&mut self) -> &mut impl DatabaseRw<PrunedTxBlobs> {
+        &mut self.7
+    }
+    fn prunable_hashes_mut(&mut self) -> &mut impl DatabaseRw<PrunableHashes> {
+        &mut self.8
+    }
+    fn outputs_mut(&mut self) -> &mut impl DatabaseRw<Outputs> {
+        &mut self.9
+    }
+    fn prunable_tx_blobs_mut(&mut self) -> &mut impl DatabaseRw<PrunableTxBlobs> {
+        &mut self.10
+    }
+    fn rct_outputs_mut(&mut self) -> &mut impl DatabaseRw<RctOutputs> {
+        &mut self.11
+    }
+    fn tx_ids_mut(&mut self) -> &mut impl DatabaseRw<TxIds> {
+        &mut self.12
+    }
+    fn tx_heights_mut(&mut self) -> &mut impl DatabaseRw<TxHeights> {
+        &mut self.13
+    }
+    fn tx_unlock_time_mut(&mut self) -> &mut impl DatabaseRw<TxUnlockTime> {
+        &mut self.14
+    }
+}
```

Comment:
1. Trait is defined that gives access to all tables, blanket impl is provided for `(a, tuple, containing, all, open, database, types, ...)`